### ISSUE TITLE
Support for _oidc_refresh response without ID token

### ISSUE THREAD

**Issue by adamcfraser** [couchbase/couchbase-lite-ios#1400]
To support OIDC providers (like Azure) that don't return a valid signed ID token from the refresh response, we'd like client to be able handle _refresh requests that return a valid SG session token, but no ID token.

Planned SG enhancement here:
https://github.com/couchbase/sync_gateway/issues/2013

@pasin to review the work required to support on CBL (all platforms).

**Comment by pasin** [comment]
I understand that SGW will have the cookie session id in the response of the _oidc_refresh request as well so the login process should be able to continue and finish without the id_token returns. 

However if there is no id_token returns, the current code will not update the token store so the token store will still have the old id_token, refresh_token, name, and session_id (session_id will not get used directly). I guessed it's OK. To be sure, we will need to test the flow after couchbase/sync_gateway#2013 is done.

**Comment by pasin** [comment]
@adamcfraser is there a way (either with configuration or a param set to the Auth URL sent to SGW to authenticate) to set the test provider to not return id_token in the result of _oidc_refresh request so that we can write a unit test or test this flow without hitting azure.

**Comment by adamcfraser** [comment]
That's already been added as requirement on https://github.com/couchbase/sync_gateway/issues/2013 - I'll let you know when that's implemented so you can grab it.

**Comment by pasin** [comment]
I verified that iOS client already supports the flow. The verification was done with Grociery-Sync-iOS - openid test app: The following steps were used to verified:

1. Setup SGW with OIDC Test Provider. Added `"unsigned_id_token": true` to the `oidc_test_provider` configuration in the `unsupported` section.

2. Setup grocery-sync bucket on the Couchbase Server and configure SGW pointing to the bucket.

3. Run Grocery-Sync app and login to the OIDC Test Provider. Type a username and verify that the user can successfully login.

4. Close the app.

5. Delete the user session doc from Couchbase Server.

6. Run the Grocery-Sync app and login again. Make sure that the same user can login without retyping the username.

7. Make sure that the refresh flow is executed by checking SGW log.
